### DISCUSSION TITLE
Runtimeerror in train loop.

### DISCUSSION THREAD

**Discussion by GAuravY19** [mrdbourke/pytorch-deep-learning#910] [Q&A] [answered]
This is my train loop:- 

```

epochs = 10

for epoch in range(epochs):
    model1.train()
    
    y_pred = model1(X_train)
    y_pred_act = torch.argmax(torch.sigmoid(y_pred), dim = 1).float()
    
    loss = loss_fn(y_pred_act, Y_train)
    
    optimizer.zero_grad()
    
    loss.backward()
    
    optimizer.step()
    
    print("LOss", loss)


```

What should I do to handle this error.

```
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```

**Comment by sametucuncu** [comment, marked as answer]
The loss is the error between expected output of the model (Y_train) and model's prediction which is y_pred in your case (y_pred = model1(X_train)). 

_y_pred_act = torch.argmax(torch.sigmoid(y_pred), dim = 1).float()_ is calculating the label from predicted value of the model. It is not exact output of the model. So, you should use output of the model while calculating the loss. So loss should be:

loss = loss_fn(y_pred, Y_train)
